### PR TITLE
Lakhveer/lifecycle handlers update

### PR DESCRIPTION
**Pull request by lakhveerk**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Summarize the changes, including the goals and reasons for this change. Be sure to call out any technical or behavior changes that reviewers should be aware of.

In [PR1878](https://github.com/OfficeDev/microsoft-teams-library-js/pull/1878), two handlers - beforeSuspendOrTermiante and onResume were added in app.lifecycle capability. They were updated versions of old teamsCore handlers - beforeUnload and onLoadHandler. Before, we were using different strings to communicate to the host sdk that the handlers are registered. This would cause issues in scenarios like: If the user has registered for both pair of handlers, what pair will be called when the user navigate away or navigate to an app? 

In this PR, I have update the handlers in app.lifecycle so that they will use the strings `beforeUnload` and `load` to talk to hostSDK. If the user register for both pair of handlers, only the lifecycle.beforeSuspendOrTerminate and lifecycle.resume will be called. Since the teamsCore handlers are marked beta, this is a minor change. 

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. Update the handlers in app.lifecycle so that they will use the strings `beforeUnload` and `load` to talk to hostSDK
2. Deleted deprecated tag from old handlers because new ones are not working yet. 

### Validation performed:

1. <Step 1>
2. <Step 2>

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>

### Related PRs:

> Remove this section if n/a

### Next/remaining steps:

> List the next or remaining steps in implementing the overall feature in subsequent PRs (or is the feature 100% complete after this?).

> Remove this section if n/a

- [ ] Item 1
- [ ] Item 2

### Screenshots:

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| < image1 > | < image2 > |

### CODE REVIEW

File: packages/teams-js/src/public/teamsAPIs.ts [hunk 1]
```diff
@@ -53,9 +53,6 @@ export namespace teamsCore {
    *
    * @param handler - The handler to invoke when the page is loaded.
    *
-   * @deprecated
-   * As of 2.14.1, please use {@link app.lifecycle.registerOnResumeHandler} instead.
-   *
```
Comment (AE-MS) [inline packages/teams-js/src/public/teamsAPIs.ts:58]: Is this no longer deprecated? I understood from the PR description that you want to maintain the same messages across the TJS->Host boundary (for compatibility and confusion reasons), but presumably we would still prefer developers not add new uses of the functions in this file? Or is that not true?

**Comment by AE-MS** [comment]
> /* eslint-disable @typescript-eslint/ban-types */

It seems like the intent, from your PR description, is to ensure that both the old and new handlers use the same message when communicating to the host.

The changes in this file seem more extensive than just updating the message -- can you say more about the motivation for that?

---
Refers to: packages/teams-js/src/internal/handlers.ts:1 in c6b7f14. [](commit_id = c6b7f14c695393eaf68267c0ed76430b109778ba, deletion_comment = False)

**Review by AE-MS — CHANGES_REQUESTED** [review]
:clock1:

**Comment by lakhveerk** [comment]
> > /* eslint-disable @typescript-eslint/ban-types */
> 
> It seems like the intent, from your PR description, is to ensure that both the old and new handlers use the same message when communicating to the host.
> 
> The changes in this file seem more extensive than just updating the message -- can you say more about the motivation for that?
> 
> Refers to: packages/teams-js/src/internal/handlers.ts:1 in [c6b7f14](https://github.com/OfficeDev/microsoft-teams-library-js/commit/c6b7f14c695393eaf68267c0ed76430b109778ba). [](commit_id = [c6b7f14](https://github.com/OfficeDev/microsoft-teams-library-js/commit/c6b7f14c695393eaf68267c0ed76430b109778ba), deletion_comment = False)

This commit opens this code
image
where an internal function is renamed to `readyToUnload` from  `readyToSuspendOrTerminate`. This is not customer facing function. So, name change would not matter much. 'readyToUnload` matches with `beforeUnload`, and that is the reason of this particular change. What do you think is extensive about this change? 

The more extensive change is If the user register for both pair of handlers, only the lifecycle.beforeSuspendOrTerminate and lifecycle.resume will be called. Since the teamsCore handlers are marked beta, I believe this is still a minor change even though it can actually be a breaking one if the developer registers a different handler for beforeSuspendOrTermiante than what they have in beforeUnload.

**Review by alexneyman-MSFT — APPROVED** [review]

**Review by AE-MS — APPROVED** [review]
:shipit:

**Pull request merged**